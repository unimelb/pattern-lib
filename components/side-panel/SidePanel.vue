<template>
  <nav class="side-panel">
    <div class="side-panel__inner">
      <h5 class="side-panel__title">
        {{ title }}
        <button
          :aria-label="title"
          class="side-panel__trigger"
          @click="toggleOpen">
          <SvgIcon
            width="10"
            height="10"
            name="close" />
        </button>
      </h5>
      <ul
        :class="{ 'side-panel__nav--closed': !open }"
        class="side-panel__nav">
        <slot />
      </ul>
    </div>
  </nav>
</template>

<script>
// side-panel-toggle-open
import SvgIcon from '../icons/SvgIcon.vue';

export default {
  components: {
    SvgIcon,
  },
  props: {
    right: {
      type: Boolean,
      default: false,
    },
    title: {
      type: String,
      default: '',
      required: true,
    },
    initialOpen: {
      type: Boolean,
      default: true,
    },
  },
  data() {
    return { open: this.initialOpen };
  },
  methods: {
    toggleOpen() {
      this.open = !this.open;
      this.$emit('side-panel-toggle-open', this.open);
    },
  },
};
</script>
