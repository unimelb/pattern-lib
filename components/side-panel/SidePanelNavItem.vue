<template>
  <li class="side-panel__nav-item clearfix">
    <a
      v-scroll-to="target"
      :href="target">
      <span v-if="right">
        <slot />
      </span>
      <span
        v-if="!right"
        class="push-icon">
        <slot />
        <SvgIcon
          class="push-icon__icon"
          name="chevron-right"
          width="10"
          height="10" />
      </span>
    </a>
  </li>
</template>

<script>
import SvgIcon from '../icons/SvgIcon.vue';

export default {
  components: {
    SvgIcon,
  },
  props: {
    target: {
      type: String,
      default: '',
    },
    right: {
      type: Boolean,
      default: false,
    },
  },
};
</script>
