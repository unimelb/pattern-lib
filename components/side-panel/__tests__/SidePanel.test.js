import {
  mount,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import SidePanel from '../SidePanel.vue';
import SidePanelNavItem from '../SidePanelNavItem.vue';

expect.extend(toHaveNoViolations);

describe('SidePanel', () => {
  it('should match snapshot', () => {
    const result = mount(SidePanel, {
      propsData: {
        title: 'mock title',
      },
      slots: { SidePanelNavItem },
    }).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = mount(SidePanel, {
      propsData: {
        title: 'mock title',
      },
      slots: { SidePanelNavItem },
    }).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
