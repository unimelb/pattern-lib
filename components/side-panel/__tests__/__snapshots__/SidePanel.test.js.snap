// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidePanel should match snapshot 1`] = `
<nav
  class="side-panel"
>
  <div
    class="side-panel__inner"
  >
    <h5
      class="side-panel__title"
    >
      
      mock title
      
      <button
        aria-label="mock title"
        class="side-panel__trigger"
      >
        <svg
          aria-label="close"
          focusable="false"
          height="10"
          role="img"
          width="10"
        >
          <use
            xlink:href="#icon-close"
          />
           
          <title>
            Close
          </title>
           
        </svg>
      </button>
    </h5>
     
    <ul
      class="side-panel__nav"
    />
  </div>
</nav>
`;
